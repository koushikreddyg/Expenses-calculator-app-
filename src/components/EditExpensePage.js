import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { selectEditExpense, setRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit1 = (expense) => {
    this.props.selectEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.setRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  //buttonclick:
  render() {
    return (
      <div>
      
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit1}
        />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  selectEditExpense: (id, expense) => dispatch(selectEditExpense(id, expense)),
  setRemoveExpense: (data) => dispatch(setRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
