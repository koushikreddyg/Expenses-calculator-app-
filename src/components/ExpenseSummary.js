import React from 'react';
import {connect} from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import selectorExpenses from '../selectors/expenses';
import numeral from 'numeral';
export const ExpenseSummary =({expensesCount,expensesTotal})=>{
    let expenseWord=expensesCount<2?'expense':'expenses'
    return(
    <div>
        <h1>Viewing {expensesCount} {expenseWord} totalling  {numeral(expensesTotal/100).format('$0,0.00')}</h1>
        
    </div>
    );
}
const mapStateToProps=(state)=>({
expensesCount:selectorExpenses(state.expenses,state.filters).length,
expensesTotal:expensesTotal(selectorExpenses(state.expenses,state.filters))
})
export default connect(mapStateToProps)(ExpenseSummary)