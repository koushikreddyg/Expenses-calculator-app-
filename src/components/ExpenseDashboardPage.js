import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesTotal from '../selectors/expenses-total';
import ExpenseSummary from './ExpenseSummary';
//import numeral from 'numeral';
//import {connect} from 'react-redux';
//import selectExpenses from '../selectors/expenses';
 const ExpenseDashboardPage = (props) => (
  <div>
  <ExpenseSummary/>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
