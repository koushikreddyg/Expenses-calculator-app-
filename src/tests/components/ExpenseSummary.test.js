import React from 'react';
import {ExpenseSummary} from '../../components/ExpenseSummary';
import expensesTotal from '../../selectors/expenses-total';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
test('test case for snapshot of this component',()=>{
    //const expensesCount=expenses.length;
    //const expensesTotal1=expensesTotal(expenses);
    const wrapper=shallow(<ExpenseSummary expensesCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot()
})
test('test case for snapshot for single expense',()=>{
    //const expensesCount=expenses.length;
    //const expensesTotal1=expensesTotal(expenses);
    const wrapper=shallow(<ExpenseSummary expensesCount={3} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot()
})