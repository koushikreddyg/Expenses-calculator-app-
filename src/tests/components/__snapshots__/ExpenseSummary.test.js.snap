// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test case for snapshot for single expense 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling  
    $2.35
  </h1>
</div>
`;

exports[`test case for snapshot of this component 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling  
    $2.35
  </h1>
</div>
`;
